client/client.go

```diff
@@ -180,21 +180,33 @@ func (e AuthorizationError) Error() string {
 	return fmt.Sprintf("cannot add authorization: %v", e.error)
 }
 
-type ConnectionError struct{ error }
+type ConnectionError struct{ Err error }
 
 func (e ConnectionError) Error() string {
 	var errStr string
-	switch e.error {
+	switch e.Err {
 	case context.DeadlineExceeded:
 		errStr = "timeout exceeded while waiting for response"
 	case context.Canceled:
 		errStr = "request canceled"
 	default:
-		errStr = e.error.Error()
+		errStr = e.Err.Error()
 	}
 	return fmt.Sprintf("cannot communicate with server: %s", errStr)
 }
 
+func (e ConnectionError) Is(err error) bool {
+	return IsConnectionError(err)
```

Comment:
I think this implementation of `Is` is incorrect: it's supposed to return `true` if any error in the chain matches the target, and it only checks the outermost one.
I think you need to hand it back to `xerrors.Is`; something like
```go
if IsConnectionError(err) {
        return true
}
return xerrors.Is(e.Err, err)
```

Reply:
staring at the code in go1.13 and xerrors (the docs are not that great) it seems the natural thing is not a recursive definition of IS but to define Unwrap on ConnectionError

Reply:
Ah, it's true that the thing that talks about the chain is the toplevel `Is`, and the docs don't say much about the interface beyond the single case.
And I can confirm having Is + Unwrap lets my silly tests pass

Reply:
Yes, sorry, this is silly! I removed this and just added an Unwrap() error

Reply:
I found https://github.com/golang/go/wiki/ErrorValueFAQ helpful fwiw.